cellprofiler/library/functions/object_processing.py

```diff
@@ -221,6 +229,180 @@ def segment_objects(labels_x, labels_y, dimensions):
 
     return output
 
+
+def watershed(
+    input_image: numpy.ndarray,
+    method: Literal["distance", "intensity", "markers"] = "distance",
+    declump_method: Literal["shape", "intensity", "none"] = "shape",
+    seed_method: Literal["local", "regional"] = "local",
+    intensity_image: numpy.ndarray = None,
+    markers_image: numpy.ndarray = None,
+    mask: numpy.ndarray = None,
+    max_seeds: int = -1,
+    downsample: int = 1,
+    min_distance: int = 1,
+    min_intensity: float = 0,
+    footprint: int = 8,
+    connectivity: int = 1,
+    compactness: float = 0.0,
+    exclude_border: bool = False,
+    watershed_line: bool = False,
+    gaussian_sigma: float = 0.0,
+    structuring_element: Literal[
+        "ball", "cube", "diamond", "disk", "octahedron", "square", "star"
+    ] = "disk",
+    structuring_element_size: int = 1,
+    return_seeds: bool = False,
+):
+    # Check inputs
+    if not numpy.array_equal(input_image, input_image.astype(bool)):
+        raise ValueError("Watershed expects a thresholded image as input")
+    if (
+        method.casefold() == "intensity" or declump_method.casefold() == "intensity"
+    ) and intensity_image is None:
+        raise ValueError(f"Intensity-based methods require an intensity image")
+
+    if method.casefold() == "markers" and markers_image is None:
+        raise ValueError("Markers watershed method require a markers image")
+
+    # Create and check structuring element for seed dilation
+    strel = getattr(skimage.morphology, structuring_element.casefold())(
+        structuring_element_size
+    )
+
+    if strel.ndim != input_image.ndim:
+        raise ValueError(
+            "Structuring element does not match object dimensions: "
+            "{} != {}".format(strel.ndim, input_image.ndim)
+        )
+
+    if input_image.ndim == 3:
+        footprint = numpy.ones((footprint, footprint, footprint))
```

Comment:
Let's not overwrite a setting value with another thing with the same name, please!